dask_image/ndinterp/__init__.py

```diff
@@ -247,6 +250,207 @@ def affine_transform(
+        ‘nearest’ (a a a a | a b c d | d d d d)
+            The input is extended by replicating the last pixel.
+    cval : scalar, optional
+        Value to fill past edges of input if mode is ‘constant’. Default is 0.0.
+    prefilter : bool, optional
+        currently not supported
+    output_shape : tuple of ints, optional
+        The shape of the array to be returned.
+    output_chunks : tuple of ints, optional
+        The shape of the chunks of the output Dask Array.
+
+    Returns
+    -------
+    rotate : Dask Array
+        A dask array representing the rotated input.
+
+    Notes
+    -----
+        Differences to `ndimage.affine_transformation`:
+        - currently, prefiltering is not supported
+          (affecting the output in case of interpolation `order > 1`)
+        - default order is 1
+        - modes 'reflect', 'mirror' and 'wrap' are not supported
+        - passing array to `output` is not currently supported.
+
+        Arguments equal to `ndimage.affine_rotate`,
+        except for `output_chunks`, `output_shape`.
+
+    Examples
+    --------
+    >>> from scipy import ndimage, misc
+    >>> import matplotlib.pyplot as plt
+    >>> import dask.array as da
+    >>> fig = plt.figure(figsize=(10, 3))
+    >>> ax1, ax2, ax3 = fig.subplots(1, 3)
+    >>> img = da.from_array(misc.ascent(),chunks=(64,64))
+    >>> img_45 = dask_image.ndinterp.rotate(img, 45, reshape=False)
+    >>> full_img_45 = dask_image.ndinterp.rotate(img, 45, reshape=True)
+    >>> ax1.imshow(img, cmap='gray')
+    >>> ax1.set_axis_off()
+    >>> ax2.imshow(img_45, cmap='gray')
+    >>> ax2.set_axis_off()
+    >>> ax3.imshow(full_img_45, cmap='gray')
+    >>> ax3.set_axis_off()
+    >>> fig.set_tight_layout(True)
+    >>> plt.show()
+    >>> print(img.shape)
+    (512, 512)
+    >>> print(img_45.shape)
+    (512, 512)
+    >>> print(full_img_45.shape)
+    (724, 724)
+
+    """
+    if not type(input_arr) == da.core.Array:
+        input_arr = da.from_array(input_arr)
+
+    if output_chunks is None:
+        output_chunks = input_arr.chunksize
+
+    if reshape & (output_shape is not None):
+        warnings.warn('Both reshaping desired and output_shape provided.'
+                      'Will use the explicit output_shape.', UserWarning)
+
+    if prefilter:
+        warnings.warn('Prefilter currently unsupported.', UserWarning)
+
+    if output_shape is None:
+        output_shape = np.asarray(input_arr.shape)
+
+    ndim = input_arr.ndim
+
+    if ndim < 2:
+        raise ValueError('input array should be at least 2D')
+
+    axes = list(axes)
+
+    if len(axes) != 2:
+        raise ValueError('axes should contain exactly two values')
+
+    if not all([isinstance(ax, Number) for ax in axes]):
+        raise TypeError('axes should contain only integer values')
```

Comment:
Is there a specific reason for why you extend here on the `scipy.ndimage.rotate` implementation?

https://github.com/scipy/scipy/blob/f990b1d2471748c79bc4260baf8923db0a5248af/scipy/ndimage/_interpolation.py#L901